lib/glific/flows/contact_setting.ex

```diff
@@ -41,13 +42,35 @@ defmodule Glific.Flows.ContactSetting do
             }
           })
 
+        # update profile languange if active profile is set for a contact
+        maybe_update_profile_language(contact, language.id)
+
         Map.put(context, :contact, contact)
 
       [] ->
         raise("Error! No language found with label #{inspect(language)}")
     end
   end
 
+  @doc """
+  Update profile language if there is an active profile id set
+  """
+  @spec maybe_update_profile_language(Contact.t(), non_neg_integer()) ::
+          Contact.t()
+  def maybe_update_profile_language(
+        %{active_profile_id: active_profile_id} = contact,
+        language_id
+      )
+      when is_integer(active_profile_id) do
+    with profile <- Profiles.get_profile!(active_profile_id) do
+      Profiles.update_profile(profile, %{language_id: language_id})
+    end
```

Comment:
Looks like you are in love with the **with**  statement. 😊

I could not find any difference between the below code
 ```
 Profiles.get_profile!(active_profile_id)
 |> Profiles.update_profile(%{language_id: language_id})
 ```
and your code.

Reply:
@pankaj-ag restructured it a bit 😇